pkgs/shells/powershell/default.nix

```diff
@@ -27,20 +27,30 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
 
   installPhase = ''
-    mkdir -p $out/bin
-    mkdir -p $out/share/powershell
-    cp -r * $out/share/powershell
-    makeWrapper $out/share/powershell/pwsh $out/bin/pwsh \
+    pslibs=$out/share/powershell
+    mkdir -p $pslibs
+
+    cp -r * $pslibs
+
+	  rm $pslibs/libcrypto.so.1.0.0
+	  rm $pslibs/libssl.so.1.0.0
+
+	  patchelf --replace-needed libcrypto.so.1.0.0 libcrypto.so.1.1 $pslibs/libmi.so
+	  patchelf --replace-needed libssl.so.1.0.0 libssl.so.1.1 $pslibs/libmi.so
```

Comment:
do you mind making this darwin compatible?

```
rm: cannot remove '/nix/store/kca66mmwd5sakpa836hmcdz1dqj260ms-powershell-7.0.0/share/powershell/libcrypto.so.1.0.0': No such file or directory
builder for '/nix/store/a3yv0rvwf11jms0i3h5466ph7kmcwfwg-powershell-7.0.0.drv' failed with exit code 1
error: build of '/nix/store/a3yv0rvwf11jms0i3h5466ph7kmcwfwg-powershell-7.0.0.drv' failed
```

Reply:
I could start by deleting those two lines but I'd be shooting in the dark afterwards...

Reply:
I wonder if I should revert to 6.2.3 for Darwin. Very foolish to have blindly changed the hash.

Reply:
I thought `autoPatchelfHook` does not exists on `macOS` at all.

Reply:
Nevertheless the patchelf code needs to be hidden behind a `optionalString !stdenv.isDarwin` flag.

Reply:
see #86423